micro-rdk-server/native/main.rs

```diff
@@ -5,39 +5,46 @@ mod native {
     #[allow(dead_code)]
     const ROBOT_SECRET: Option<&str> = option_env!("MICRO_RDK_ROBOT_SECRET");
 
-    use micro_rdk::common::{
-        conn::network::ExternallyManagedNetwork, credentials_storage::RAMStorage,
-        entry::RobotRepresentation,
+    use micro_rdk::{
+        common::{
+            conn::network::ExternallyManagedNetwork, credentials_storage::RAMStorage,
+            entry::RobotRepresentation,
+        },
+        native::entry::serve_web_with_external_network,
     };
-    use micro_rdk::native::entry::serve_web_with_external_network;
 
+    #[allow(unreachable_code)]
     pub(crate) fn main_native() {
         env_logger::builder()
             .format_timestamp(Some(env_logger::TimestampPrecision::Millis))
             .init();
 
+        #[allow(unused)]
         let repr = RobotRepresentation::WithRegistry(Box::default());
 
+        #[allow(unused)]
         let network = match local_ip_address::local_ip().expect("error parsing local IP") {
             std::net::IpAddr::V4(ip) => ExternallyManagedNetwork::new(ip),
             _ => panic!("oops expected ipv4"),
         };
-        #[cfg(has_robot_config)]
-        {
-            if ROBOT_ID.is_some() && ROBOT_SECRET.is_some() {
-                let ram_storage = RAMStorage::new("", "", ROBOT_ID.unwrap(), ROBOT_SECRET.unwrap());
-                serve_web_with_external_network(None, repr, 3, ram_storage, network);
-            }
-        }
-        #[cfg(not(has_robot_config))]
+
+        let storage = if cfg!(has_robot_config) && ROBOT_ID.is_some() && ROBOT_SECRET.is_some() {
+            RAMStorage::new("", "", ROBOT_ID.unwrap(), ROBOT_SECRET.unwrap())
```

Comment:
I can change build.rs so that `has_robot_config` is only dependent on `viam.json`, not wifi credentials. Then for `esp32`, I can set the wifi credentials only if the static options of SSID/PASSWORD are not none, in lieu of a `has_wifi_creds` cfg flag.